notification/store.go

```diff
@@ -280,3 +295,45 @@ func (db *DB) VerifyContactMethod(ctx context.Context, cmID string, code int) er
 
 	return nil
 }
+
+func (db *DB) FindManyMessageStatuses(ctx context.Context, ids ...string) ([]MessageStatus, error) {
+	err := permission.LimitCheckAny(ctx, permission.User, permission.Admin)
```

Comment:
Can just be a `User` check, as `Admin` will automatically qualify (some old code references both, but it's unnecessary with the current permissions package).